elastic/detection-rules
# Validate against beats and integrations schemas

## Issues

#1994 

## Summary
- Follow up to https://github.com/elastic/detection-rules/pull/2470 to always validate against ecs and then beats, followed by optionally validating against integration schemas vs prior which validated beats or integrations.
- Small refactor for readability.
- Only warns for unit tests.

## Testing

During validation rules that have both integrations and beats indexes should validate against both the integration schema and the appropriate beats. 

- Run `view-rule` on several different integrations and try with multiple `min_stack` versions.  
  - E.g. `python -m detection_rules view-rule rules/integrations/google_workspace/collection_google_drive_ownership_transferred_via_google_workspace.toml` 
  - E.g. `python -m detection_rules view-rule rules/integrations/aws/collection_cloudtrail_logging_created.toml` 
  - E.g. `python -m detection_rules view-rule rules/integrations/azure/initial_access_consent_grant_attack_via_azure_registered_application.toml` 
  - E.g. `python -m detection_rules view-rule rules/apm/apm_403_response_to_a_post.toml` 
- Run `show-latest-compatible` to verify command shows latest compatible version.  
  - E.g. `python -m detection_rules dev integrations show-latest-compatible -p apm -s 8.2.0`
-  Run the `build-schemas` and `build-manifest` commands to test rebuilding the schemas and manifest
   - E.g. `python -m detection_rules dev integrations build-schemas -o`
   - E.g. `python -m detection_rules dev integrations build-manifests -o`
- Run `make test` to ensure all unit tests pass.
